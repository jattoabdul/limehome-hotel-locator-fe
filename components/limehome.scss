* {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  box-sizing: border-box;
}

.app-wrapper {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.list-wrapper {
  padding: 1rem;
  padding-top: 10rem;
  position: relative;
}

.res {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  background: #ffffff;
  padding: 0 1rem;
}

.header-wrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.header-wrapper img {
  width: 3rem;
}

.header-wrapper .hambuger {
  position: relative;

  &, &:before, &:after {
    background: #000000;
    width: 2rem;
    height: 2px;
  }

  &:before, &:after {
   content: ' ';
   position: absolute;
   left: 0;
  }

  &:before {
    top: -0.5rem;
   }

  &:after {
    bottom: -0.5rem;
  }
}

form {
  display: flex;
  align-items: center;
  margin-top: 2rem;

  input[type="search"] {
    outline: none;
    border: #f7f7f7;
    font-size: 1.75rem;
    padding: .5rem;
  }

  input[type="submit"] {
    margin-left: .5rem;
    border: none;
    color: white;
    background: #00f2a9;
    padding: .5rem 1rem;
    cursor: pointer;

    &:focus {
      outline: none;
    }
  }
}

.hotel-result-list {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: 5%;
  grid-row-gap: 1.5rem;
  padding: 1rem 2.5%;
  height: calc(100vh - 12rem);
  overflow-y: auto;
  grid-auto-flow: dense;

  /* width */
  &::-webkit-scrollbar {
    width: 2px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: #f1f1f1;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: #00f2a9;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: #014c35;
  }
}

.hotel-result-item {
  img {
    width: 100%;
    border-radius: 3px;
  }

  h2 {
    font-size: 1rem;
    font-weight: 500;
    line-height: 1.25;
    margin: 0.5rem 0;
  }
  p {
    font-size: 0.75rem;
    font-weight: 400;
    margin: 0;
  }
}

.map-wrapper {}

