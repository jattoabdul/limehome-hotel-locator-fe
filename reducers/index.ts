import { combineReducers } from 'redux';
import limeHome, { initialState as limeHomeState } from './limehome';

export const initialState = {
  limeHome: limeHomeState
};

export default combineReducers({
  limeHome
});
