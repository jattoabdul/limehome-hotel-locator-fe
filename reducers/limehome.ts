import {
  LOCATION_ONCHANGE,
  PROXIMITY_ONCHANGE,
  HOTELS_SET,
  HOTELS_CLEAR,
} from '../constants/actionTypes';

export const initialState = {
  location: {
    value: '',
  },
  proximity: {
    value: '',
  },
  hotels: [],
  data: [],
};

export default (state = initialState, action) => {
  const {
    type,
    location,
    proximity,
    hotels
  } = action;

  switch (type) {
    case LOCATION_ONCHANGE: {
      return Object.assign({}, state, {
        location,
      });
    }
    case PROXIMITY_ONCHANGE: {
      return Object.assign({}, state, {
        proximity,
      });
    }

    case HOTELS_SET: {
      // TODO: check to be sure hotels is an array
      return Object.assign({}, state, {
        hotels: [],
        data: [...hotels],
      });
    }

    case HOTELS_CLEAR: {
      return Object.assign({}, state, {
        data: [],
      });
    }

    default: {
      return state;
    }
  }
};
