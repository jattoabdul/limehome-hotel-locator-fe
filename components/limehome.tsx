import React from 'react';
import { compose } from 'recompose';
import getConfig from "next/config";
import { usePosition } from 'use-position';
import SimpleMap from './map';
import './limehome.scss';

const { publicRuntimeConfig } = getConfig();

const LimeHome= (props) => {
  const {
    location,
    proximity,
    data,
    onChangeLocation,
    onChangeProximity,
    setHotels,
    clearHotels,
  } = props;

  const mapStyles = {height: '100vh', width: '50vw',};

  // TODO: tune properly to get accurate current location
  const { latitude, longitude, error } = usePosition(true, {watch: true, enableHighAccuracy: true});
  let mapCenter;
  if (location.value === '' && latitude && longitude && !error) {
    onChangeLocation({
      value: `${latitude},${longitude}`,
    })
  }
  if (location.value !== '') {
    mapCenter = {lat: Number(location.value.split(',')[0]), lng: Number(location.value.split(',')[1])};
  } else {
    onChangeLocation({
      value: '6.522500,3.321350',
    })
    mapCenter = {lat: 6.522500, lng: 3.321350}; // fallback to my home co-ordinates as center
  }
  
  return (
    <React.Fragment>
      <div className="app-wrapper">
        <div>
          <h1>LimeHome</h1>
          <form onSubmit={(e) => {
            e.preventDefault();
            setHotels({
              apiKey: publicRuntimeConfig.gApiKey,
              location:  location.value,
              proximity: proximity.value,
              opennow: true
            });
          }}>
            <input
              type="text"
              value={location.value}
              onChange={e => onChangeLocation({
                value: e.target.value,
              })}
            />
            <br />
            <input
              type="submit"
              value="Locate Nearby Hotels"
            />
          </form>
          {/* proximity - input field */}
          {data.map((hotel, index) => (
            <p key={index}>
              {hotel.name}
            </p>
          ))}
        </div>
        <div>
          <SimpleMap mapStyles={mapStyles} locations={data} mapCenter={mapCenter} zoomRate={13} />
        </div>
      </div>
    </React.Fragment>
  );
};

// apiKey: 'AIzaSyBfE6ygYzCDsyohOgUlTj-lN4Npx4GYoKU'
export default compose()(LimeHome);
