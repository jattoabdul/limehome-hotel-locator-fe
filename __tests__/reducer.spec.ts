import reducer from '../reducers/limehome'
import { initialState } from '../reducers/limehome'

describe('limehome reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle LOCATION_ONCHANGE', () => {
    expect(
      reducer(initialState,
      {
        type: 'LOCATION_ONCHANGE',
        value: { value: '6.522500,3.321350' }
      })
    ).toMatchSnapshot()
  })
})
