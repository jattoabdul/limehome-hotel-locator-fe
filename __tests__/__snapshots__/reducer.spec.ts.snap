// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`limehome reducer should handle LOCATION_ONCHANGE 1`] = `
Object {
  "data": Array [],
  "hotels": Array [],
  "location": undefined,
  "mapCenter": Object {
    "value": Object {},
  },
  "proximity": Object {
    "value": "",
  },
}
`;

exports[`limehome reducer should return the initial state 1`] = `
Object {
  "data": Array [],
  "hotels": Array [],
  "location": Object {
    "value": "",
  },
  "mapCenter": Object {
    "value": Object {},
  },
  "proximity": Object {
    "value": "",
  },
}
`;
